test/refract/data/test-Member.cc

```diff
@@ -0,0 +1,215 @@
+                THEN("nothing was called on the mock")
+                {
+                    REQUIRE(test::ElementMock::last_instance()._total_ctx == 0);
+                }
+            }
+
+            WHEN("it goes out of scope")
+            {
+                THEN("the value is destroyed")
+                {
+                    REQUIRE(test::ElementMock::instances().empty());
+                }
+            }
+        }
+    }
+
+    GIVEN("A member_t with a StringElement key and an ElementMock value")
+    {
+        std::string key = "eigenvalue";
+        auto mock = new test::ElementMock{};
+        mock->clone_out = std::make_unique<test::ElementMock>();
+        auto clone_out_ptr = mock->clone_out.get();
+        mock->_value = 349802;
+
+        REQUIRE(test::ElementMock::instances().size() == 2);
+        member_t member(key, std::unique_ptr<IElement>(mock));
+
+        WHEN("another member_t is copy constructed from it")
+        {
+            member_t member2(member);
+
+            THEN("its key and value differ from the original")
+            {
+                REQUIRE(member.key() != member2.key());
+                REQUIRE(member.value() != member2.value());
+            }
+
+            THEN("its key is a StringElement containing a copy of the original key string")
+            {
+                auto strValue = dynamic_cast<const StringElement*>(member2.key());
+                REQUIRE(strValue);
+                REQUIRE(key == strValue->get());
+            }
+
+            THEN("its value is the result of invoking clone() on original value")
+            {
+                REQUIRE(mock->_total_ctx == 1);
+                REQUIRE(mock->clone_ctx == 1);
+                REQUIRE(mock->clone_in == IElement::cAll);
+
+                REQUIRE(member2.value() == clone_out_ptr);
+            }
+
+            THEN("no further ElementMocks are constructed")
+            {
+                REQUIRE(test::ElementMock::instances().size() == 2);
+            }
+        }
+
+        WHEN("another member_t is move constructed from it")
+        {
+            auto memberKey = member.key();
+            auto memberValue = member.value();
+
+            member_t member2(std::move(member));
+
+            THEN("the original's key and value are now nullptr")
+            {
+                REQUIRE(member.key() == nullptr);
+                REQUIRE(member.value() == nullptr);
+            }
+
+            THEN("its key and value are equal to the original key and value")
+            {
+                REQUIRE(member2.key() == memberKey);
+                REQUIRE(member2.value() == memberValue);
+            }
+
+            THEN("nothing gets called on the original mock")
+            {
+                REQUIRE(mock->_total_ctx == 0);
+            }
+
+            THEN("no further ElementMocks are constructed")
+            {
+                REQUIRE(test::ElementMock::instances().size() == 2);
+            }
+        }
+    }
```

Comment:
It seems (by interface) we support change of key value
add test for this case
if not supported, we should add test case to avoid

Reply:
removed from implementation, wasn't used